src/core/public/chrome/nav_controls/nav_controls_service.ts

```diff
@@ -64,6 +64,8 @@ export interface ChromeNavControls {
   registerCenter(navControl: ChromeNavControl): void;
   /** Register a nav control to be presented on the left-bottom side of the left navigation. */
   registerLeftBottom(navControl: ChromeNavControl): void;
+  /** Register a nav control to be presented on the right side of the new primary chrome header. */
```

Comment:
what does the "new primary chrome header" refer to?

Reply:
So we're trying to introduce an API to register a global control button which is available for all application headers and page headers. Can we simply call it `registerHeaderRight`?

Reply:
I'm open to renaming it to registerHeaderRight. I just want to add some explanations for my naming convention. I added 'New' to distinguish it from the legacy header since this slot won't be displayed in that header. I added 'Primary' to distinguish it from the secondary header, as it's positioned on the right side of the primary header. Do you think this naming convention is reasonable? Using 'headerRight' is simpler, but I'm not sure if it might confuse others into thinking that the registered elements won't be displayed.

Reply:
Thanks for the explanation! I got the point of "Primary" now :)
For "New", since legacy header is meant to be deprecated, so I don't feel necessary to distinguish it from the legacy one, imagine in the future, the legacy header was removed, then the "New" will sound a bit strange. what about calling it `registerPrimaryHeaderRight`?